ledger/src/blockstore.rs

```diff
@@ -2915,46 +2918,68 @@ impl Blockstore {
     }
 
     /// Gets accounts used in transactions in the slot range [starting_slot, ending_slot].
+    /// Additionally returns a bool indicating if the set may be incomplete.
     /// Used by ledger-tool to create a minimized snapshot
     pub fn get_accounts_used_in_range(
         &self,
         bank: &Bank,
         starting_slot: Slot,
         ending_slot: Slot,
-    ) -> DashSet<Pubkey> {
+    ) -> (DashSet<Pubkey>, bool) {
         let result = DashSet::new();
+        let lookup_tables = DashSet::new();
+        let possible_cpi_alt_extend = AtomicBool::new(false);
+
+        fn add_to_set<'a>(set: &DashSet<Pubkey>, iter: impl IntoIterator<Item = &'a Pubkey>) {
```

Comment:
nice refactoring.

nit: you can create closure not to pass around `result` maybe? like:

```rust
let add_to_set = |iter: ...| { ... result.insert(...) ... } 
```

Reply:
oh, this is used for `result` and `lookup_tables`.. never mind...